src/status.rs

```diff
@@ -0,0 +1,95 @@
+//! A Status Report announces the station's current mission or any other single
+//! line status to everyone. The report starts with the '>' APRS Data Type Identifier.
+//! The report may optionally contain a timestamp.
+//!
+//! Examples:
+//! - ">12.6V 0.2A 22degC"              (report without timestamp)
+//! - ">120503hFatal error"             (report with timestamp in HMS format)
+//! - ">281205zSystem will shutdown"    (report with timestamp in DHM format)
+
+use std::convert::TryFrom;
+use std::io::Write;
+
+use AprsError;
+use EncodeError;
+use Timestamp;
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct AprsStatus {
+    pub timestamp: Option<Timestamp>,
+    pub comment: Vec<u8>,
+}
+
+impl TryFrom<&[u8]> for AprsStatus {
+    type Error = AprsError;
+
+    fn try_from(b: &[u8]) -> Result<Self, Self::Error> {
+        // parse timestamp if available
+        let has_timestamp = b.len() >= 7
+            && (b[6] == b'h' || b[6] == b'z')
+            && b[0..6].iter().all(|c| c.is_ascii_digit());
+        let timestamp = if has_timestamp {
+            Some(Timestamp::try_from(
+                b.get(0..7)
+                    .ok_or_else(|| AprsError::InvalidStatus(b.to_vec()))?,
+            )?)
+        } else {
+            None
+        };
+
+        // set the comment (strip potential timestamp)
+        let comment = if has_timestamp { &b[7..] } else { &b[0..] };
```

Comment:
```suggestion
        let timestamp = b.get(..7).and_then(|b| Timestamp::try_from(b).ok());
        let comment = if timestamp.is_some() { &b[7..] } else { b };
```

This logic here really boils down to "If the first 7 bytes are a valid timestamp, interpret it as a timestamp and the rest as a comment. Otherwise, the whole thing is a comment." We've already implemented logic for determining whether something is a valid timestamp: `Timestamp::try_from`. IMO it's cleaner to just use that than re-implementing that logic here.

Reply:
Thank you, your code looks much more elegant...